numba/core/lowering.py

```diff
@@ -305,6 +305,12 @@ def create_cfunc_wrapper(self):
     def setup_function(self, fndesc):
         # Setup function
         self.function = self.context.declare_function(self.module, fndesc)
+        if self.flags.dbg_optnone:
+            attrset = self.function.attributes
+            attrset.add("optnone")
+            # optnone requires noinline unless it has alwaysinline
+            attrset.discard("alwaysinline")
+            attrset.add("noinline")
```

Comment:
It looks like this always adds `optnone noinline` and removes `alwaysinline`, should it be:
```
if not ("alwaysinline" in attrset):
    attrset.add("optnone")
    attrset.add("noinline")
```
i.e. if`alwaysinline` is present, do nothing, else add `optnone noinline`?

Reply:
applied in 9c9a56b